### PR TITLE
(#843) Add page for configuring CCM with HSTS

### LINKED ISSUE #843: Add a page for configuring HSTS for Chocolatey Central Management
## What New Or Updated Would You Like To See?

A page that covers how you might configure HSTS for Chocolatey Central Management.

## Why Is It Needed?

HSTS allows you to be more certain that your connection to a server is secure.

### PR DESCRIPTION
**Pull request by corbob**
## Description Of Changes

Add a page regarding configuring CCM with HSTS.

## Motivation and Context

Adding documentation for ways to enable HSTS for CCM.

## Testing

* [x] I have previewed these changes using the [Docker Container](https://github.com/chocolatey/docs/tree/master/.devcontainer) or another method before submitting this pull request.

## Change Types Made

* [ ] Minor documentation fix (typos etc.).
* [ ] Major documentation change (refactoring, reformatting or adding documentation to existing page).
* [x] New documentation page added.
* [ ] The change I have made should have a video added, and I have raised an issue for this.
    * Issue #

## Change Checklist

* [x] Requires a change to menu structure (top or left hand side)/
* [x] Menu structure has been updated

## Related Issue

- Fixes #843
- ENGTASKS-3338

### CODE REVIEW

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+Order: 60
+xref: configure-chocolatey-central-management-with-hsts
+Title: How To Configure Chocolatey Central Management to use HSTS
+Description: How To Configure Chocolatey Central Management to use HSTS
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:5]:
```suggestion
Title: How to Configure Chocolatey Central Management to Use HSTS
Description: How to Configure Chocolatey Central Management to Use HSTS
```

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+## Overview
+
+> :choco-warning: **WARNING**
+>
+> Although HTTP Strict Transport Security (HSTS) allows you to have reasonable certainty that your connection to Chocolatey Central Management is not intercepted, we recommend that you do not connect your Chocolatey Central Management server directly to the internet. Instead you should consider connecting through a load balancer or reverse proxy of the like.
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:12]:
```suggestion
> Although HTTP Strict Transport Security (HSTS) allows you to have reasonable certainty that your connection to Chocolatey Central Management is not intercepted, we recommend that you do not connect your Chocolatey Central Management server directly to the internet.
```

Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:12]: Lets avoid making recommendations that we don't provide additional information / configuration / setup for.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+Chocolatey Central Management is hosted by an IIS server, and can make use of the features available in IIS. Below is some general information about how to enable HSTS with IIS. You can also make use of a reverse proxy to sit between your Chocolatey Central Management server and your clients.
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:14]:
```suggestion
Chocolatey Central Management is hosted by II and makes use of features available by it. Below is some general information about how to enable HSTS with both IIS and a reverse proxy.
```

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+## Enabling HSTS Within IIS
+
+If you are running IIS 10.0 version 1709 or later, you can enable HSTS using the `<hsts>` element of the `<site>` element. Microsoft has information as well as samples [here](https://learn.microsoft.com/en-us/iis/configuration/system.applicationhost/sites/site/hsts). If you are on newer versions of IIS, you may even have access to enabling HSTS within the IIS Management Console; steps to enable it this way are below.
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:18]:
```suggestion
If you are running IIS 10.0 version 1709 or later, you can enable HSTS using the [documentation, and samples, provided by Microsoft](https://learn.microsoft.com/en-us/iis/configuration/system.applicationhost/sites/site/hsts). If you are on newer versions of IIS, you may even have access to enabling HSTS within the IIS Management Console; steps to enable it this way are below.
```

Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:18]: >  If you are on newer versions of IIS, you may even have access to enabling HSTS within the IIS Management Consol

I don't understand this. It says you may have access? What if you don't? What do they do? Is there a plugin / addin for IIS that is needed?

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+1. Open the IIS Management Console
+1. In the Connection pane, expand the server and then Sites to select `ChocolateyCentralManagement`
+1. On the right, select HSTS  under `Configure` in the Actions pane.
+1. Configure HSTS as desired.
+
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:24]:
```suggestion
1. Open the IIS Management Console.
1. In the **Connection** pane, expand the server and then **Sites** to select `ChocolateyCentralManagement`.
1. On the right, select **HSTS**  under **Configure** in the **Actions** pane.
1. Configure HSTS as desired.

```

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+### Enabling HSTS within IIS prior to IIS 10.0
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:25]:
```suggestion
### Enabling HSTS Within IIS Prior to IIS 10.0
```

Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:25]: Please be consistent with headings.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 5 above)
Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:24]: Please be consistent with ending bullet points with full stops / periods.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+Although it sounds like HSTS is not available in IIS prior to 10.0 version 1709, you may be able to configure it with a custom setup. There is [an external blog post](https://www.saotn.org/enable-http-strict-transport-security-hsts-on-iis/) that goes into the steps needed to configure this.
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:27]: > Although it sounds like
> you may be able to

This does not sound confident when reading it. Please state whether it's available or not and if you are not, advise the user how to determine for themselves.

Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:27]: This blog post is 8 years old. Is it accurate and useful for IIS 10 on a later OS?

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+### Enabling HSTS Outside of IIS
+
+In order to configure HSTS outside of IIS, you could use a reverse proxy and configure HSTS on the reverse proxy server. For instance nginx has an [article covering HSTS configuration](https://www.nginx.com/blog/http-strict-transport-security-hsts-and-nginx/), or you could use Apache and follow a guide such as [this one](https://www.simplified.guide/apache/enable-hsts).
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:31]: Can we change this to be more specific than 'Outside of IIS' and be more 'Nginx and Apache'?

**Review by pauby — CHANGES_REQUESTED** [review]

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 3 above)
Comment (corbob) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md]: There's a typo here in the suggestion. Adding a comment to remind myself to come back and edit the commit to add an S.

**Comment by steviecoaster** [comment]
The steps as outlined for IIS 10.0 are accurate and work correctly, though I really don't know what any of the settings mean, outside of "redirect http to https"

**Comment by corbob** [comment]
> The steps as outlined for IIS 10.0 are accurate and work correctly, though I really don't know what any of the settings mean, outside of "redirect http to https"

@steviecoaster Should we consider linking directly to the table of attributes [here](https://learn.microsoft.com/en-us/iis/configuration/system.applicationhost/sites/site/hsts#attributes)?

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 3 above)
Comment (corbob) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md]: I've edited the `S` into the commit. Re-resolving this conversation.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 7 above)
Comment (corbob) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md]: I've reworded this section and linked to the Microsoft article.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 8 above)
Comment (corbob) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md]: I've updated the heading and adjusted the wording of the start of the line.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 4 above)
Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md]: Okay, can you reword this then to say that if you're on later version and you don't have the GUI see the Microsoft samples?

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+## Enabling HSTS Within IIS
+
+If you are running IIS 10.0 version 1709 or later, you can enable HSTS using the [documentation, and samples, provided by Microsoft](https://learn.microsoft.com/en-us/iis/configuration/system.applicationhost/sites/site/hsts). If you are on newer versions of IIS, you may even have access to enabling HSTS within the IIS Management Console; steps to enable it this way are below.
+
+1. Open the IIS Management Console.
+1. In the **Connection** pane, expand the server and then **Sites** to select `ChocolateyCentralManagement`.
+1. On the right, select **HSTS**  under **Configure** in the **Actions** pane.
+1. Configure HSTS as desired.
+
+### Enabling HSTS Within IIS Prior to IIS 10.0
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:25]:
```suggestion
### Enabling HSTS Prior to IIS 10.0 Version 1709
```

Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:25]: I'm assuming from the text in this section, that the heading is correct?

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+While HSTS is available natively within IIS 10.0 version 1709, it is [possible although complicated](https://learn.microsoft.com/en-us/iis/get-started/whats-new-in-iis-10-version-1709/iis-10-version-1709-hsts#challenges-on-enabling-hsts-before-iis-100-version-1709) to use it with IIS prior to this version.
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:27]:
```suggestion
While HSTS is available natively within IIS 10.0 version 1709, it is [possible](https://learn.microsoft.com/en-us/iis/get-started/whats-new-in-iis-10-version-1709/iis-10-version-1709-hsts#challenges-on-enabling-hsts-before-iis-100-version-1709) to use it with IIS prior to this version.
```

Comment (pauby) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:27]: 'complicated' is a relative term. We should only providing information and facts.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+### Using a Reverse Proxy to Enable HSTS
+
+While Chocolatey Central Management runs on IIS, you could use a reverse proxy and configure HSTS on the reverse proxy server. For instance NGINX has an [article covering HSTS configuration](https://www.nginx.com/blog/http-strict-transport-security-hsts-and-nginx/), or you could use Apache and follow a guide such as [this one](https://www.simplified.guide/apache/enable-hsts).
```
Comment (pauby) [inline input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md:31]:
```suggestion
While Chocolatey Central Management runs on IIS, you could use a reverse proxy and configure HSTS on the reverse proxy server. For instance NGINX has an [article covering HSTS configuration](https://www.nginx.com/blog/http-strict-transport-security-hsts-and-nginx/), or you could use Apache and follow a guide such as [this one](https://www.simplified.guide/apache/enable-hsts).

We provide this information as a reference for enabling HSTS and you should evaluate and apply the necessary configuration for your environment. 
```

**Review by pauby — CHANGES_REQUESTED** [review]
This is looking good. Apologies for missing these.

File: input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md (hunk 4 above)
Comment (corbob) [reply input/en-us/guides/organizations/configure-chocolatey-central-management-with-hsts.md]: I had some troubles wrapping my head around the best way to reword this. I ended up settling on adding a blurb afterwards that links to the Microsoft page for how to configure it in that version.

**Comment by pauby** [comment]
> Is "if you tick this box in IIS, the Central Management Web Interface still works" good enough? 😂

I think this is a good approach to this. This has already grown arms and legs and was only intended to be a light overview of what can be done!

**Review by pauby — CHANGES_REQUESTED** [review]
Missed one. Other than this I'm happy to merge this.

**Review by pauby — APPROVED** [review]

**Comment by pauby** [comment]
@corbob awesome work! Thanks for your contribution!

**Pull request merged**